syntax = "proto2";

package language.google.fact_check;

message WikipediaDump {
  message Entity {
    optional string mention = 1;
    optional string entity = 2;
  }

  message Sentence {
    optional string text = 1;
    repeated Entity entities = 2;
  }

  optional string id = 1;
  optional string text = 2;
  optional string title = 3;
  map<int32, Sentence> sentences = 4;
}

message FeverExample {
  // Format is
  //    [Annotation ID, Evidence ID, Wikipedia URL, sentence ID]
  // see http://fever.ai/2018/task.html#TrainingDevelopment_Data_format_30
  // We do not care about the unused Annotation and Evidence IDs, and optionally
  // add `text' which represents the actual sentence contents.
  message Evidence {
    optional string wikipedia_url = 1;
    optional string sentence_id = 2;

    // Not populated in gold data.
    optional string sentence = 3;

    optional string page_title = 4;
  }

  message EvidenceSet {
    repeated Evidence evidence = 1;
  }

  enum Label {
    UNKNOWN_LABEL = 0;
    SUPPORTS = 1;
    REFUTES = 2;
    NOT_ENOUGH_INFO = 3;
  }

  optional string id = 1;
  optional Label label = 2;

  optional string claim = 3;

  repeated EvidenceSet evidences = 4;
}
